main.tf

```diff
@@ -1,64 +1,69 @@
-module "role_label" {
-  source  = "cloudposse/label/null"
-  version = "0.25.0"
-  context = module.this.context
-
-  delimiter           = coalesce(module.this.context.delimiter, "_")
-  regex_replace_chars = coalesce(module.this.context.regex_replace_chars, "/[^_a-zA-Z0-9]/")
-  label_value_case    = coalesce(module.this.context.label_value_case, "upper")
+data "context_label" "this" {
+  delimiter  = local.context_template == null ? var.naming_scheme.delimiter : null
+  properties = local.context_template == null ? var.naming_scheme.properties : null
+  template   = local.context_template
+
+  replace_chars_regex = var.naming_scheme.replace_chars_regex
+
+  values = merge(
+    var.naming_scheme.extra_labels,
+    { name = var.name }
+  )
 }
 
 resource "snowflake_account_role" "this" {
-  count = module.this.enabled ? 1 : 0
-
-  name    = local.name_from_descriptor
+  name    = data.context_label.this.rendered
```

Comment:
is there a possibility to disable subtree of modules? (I see `module.this.enabled` removed and I don't see equivalent option in provider approach 🤔)

Reply:
No - after a brief discussion with @jakubigla we decided to ditch that functionality. 
Especially since, `enabled` is now configured on the provider level, and honestly, it is hardly ever used (at least by myself)